src/Commands/IndexCategoriesCommand.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Rapidez\Core\Commands;
+
+use Rapidez\Core\Commands\ElasticsearchIndexCommand;
+use Rapidez\Core\Facades\Rapidez;
+use TorMorten\Eventy\Facades\Eventy;
+
+class IndexCategoriesCommand extends ElasticsearchIndexCommand
+{
+    protected $signature = 'rapidez:index:categories {store? : Store ID from Magento}';
+
+    protected $description = 'Index the Rapidez categories to Elasticsearch';
+
+    public function handle(): int
+    {
+        $categoryModel = config('rapidez.models.category');
+
+        $this->indexStores(
+            Rapidez::getStores($this->argument('store')),
+            'categories',
+            fn() => config('rapidez.models.category')::select((new $categoryModel)->qualifyColumns(['entity_id', 'name', 'url_path', 'children_count']))
+                ->whereNotNull('url_key')->whereNot('url_key', 'default-category')
+                ->where('children_count', '>', 0)
+                ->get() ?? [],
+            fn($data) => Eventy::filter('index.category.data', $data),
+            'entity_id'
+        );
```

Comment:
This isn't very readable, maybe change it to methods which can be chained or maybe typehint the parameters so we know what everything is. Like `'categories'`, that's the index name but I know that after looking at the `indexStores()` method.

```suggestion
        $this->indexStores(
            stores: Rapidez::getStores($this->argument('store')),
            indexName: 'categories',
            items: fn() => config('rapidez.models.category')::select((new $categoryModel)->qualifyColumns(['entity_id', 'name', 'url_path', 'children_count']))
                ->whereNotNull('url_key')->whereNot('url_key', 'default-category')
                ->where('children_count', '>', 0)
                ->get() ?? [],
            dataFilter: fn($data) => Eventy::filter('index.category.data', $data),
            id: 'entity_id'
        );
```

What do you think @indykoning?

Reply:
With regards to chaining, I don't think something like this would be that much cleaner:

```php
$this->indexItems(fn() => config('rapidez.models.category')::select((new $categoryModel)->qualifyColumns(['entity_id', 'name', 'url_path', 'children_count']))
        ->whereNotNull('url_key')->whereNot('url_key', 'default-category')
        ->where('children_count', '>', 0)
        ->get() ?? [])
    ->withStores($this->argument('store'))
    ->filter(fn($data) => Eventy::filter('index.category.data', $data))
    ->id('entity_id')
    ->doIndex('categories')
```

There's probably a slightly nicer way to structure it but this seems a bit less intuitive to me?

Reply:
True true, typed params are cleaner. Any thoughts @indykoning?

Reply:
Yeah i agree typed params may be cleaner.
I think the reason readability may be struggling a little is the shorthand function with the query as function argument as it spans multiple lines since it needs to for readablility of the query itself.

As to how i'm passing the function through see: https://www.php.net/releases/8.1/en.php#first_class_callable_syntax

```suggestion
        $this->indexStores(
            stores: Rapidez::getStores($this->argument('store')),
            indexName: 'categories',
            items: $this->getCategories(...),
            dataFilter: fn($data) => Eventy::filter('index.category.data', $data),
            id: 'entity_id'
        );
...
public function getCategories()
{
    return config('rapidez.models.category')::query()
                ->select((new config('rapidez.models.category'))->qualifyColumns(['entity_id', 'name', 'url_path', 'children_count']))
                ->whereNotNull('url_key')
                ->whereNot('url_key', 'default-category')
                ->where('children_count', '>', 0)
                ->get()
}
```

However i am seeing the folowing query used in more places. Maybe it's an idea to turn it into a scope?
```php
                ->whereNotNull('url_key')
                ->whereNot('url_key', 'default-category')
                ->where('children_count', '>', 0)
```